examples/go/go-client/terraform/resources/api_default.go

```diff
@@ -285,13 +286,9 @@ func createCredentialsAWS(ctx context.Context, d *schema.ResourceData, m interfa
 	idParts := []string{}
 	idParts = append(idParts, (*r.Sid))
 	d.SetId(strings.Join(idParts, "/"))
+	d.Set("sid", *r.Sid)
 
-	err = MarshalSchema(d, r)
```

Comment:
@childish-sambino Sam, Can you help me with this code changes? Its failing the go lang test
Error logs:
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/pmezard/go-difflib v1.0.0
{"Time":"2022-06-15T16:07:50.25721101Z","Action":"output","Package":"go-client/helper/client","Output":"?   \tgo-client/helper/client\t[no test files]\n"}
{"Time":"2022-06-15T16:07:50.257327426Z","Action":"skip","Package":"go-client/helper/client","Elapsed":0}
{"Time":"2022-06-15T16:07:51.251068093Z","Action":"run","Package":"go-client/helper/rest/api/v2010","Test":"TestGet"}
{"Time":"2022-06-15T16:07:51.251104718Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"=== RUN   TestGet\n"}
{"Time":"2022-06-15T16:07:51.25445176Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"    api_test.go:31: \n"}
{"Time":"2022-06-15T16:07:51.25447526Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"        \tError Trace:\tapi_test.go:31\n"}
{"Time":"2022-06-15T16:07:51.254478468Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"        \tError:      \tExpected nil, but got: \u0026url.Error{Op:\"Get\", URL:\"http://prism_twilio:4010/v1/Credentials/AWS\", Err:(*net.OpError)(0x4000184000)}\n"}
{"Time":"2022-06-15T16:07:51.254482052Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"        \tTest:       \tTestGet\n"}
{"Time":"2022-06-15T16:07:51.254525593Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"    api_test.go:32: \n"}
{"Time":"2022-06-15T16:07:51.254528468Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"        \tError Trace:\tapi_test.go:32\n"}
{"Time":"2022-06-15T16:07:51.254530343Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"        \tError:      \tExpected value not to be nil.\n"}
{"Time":"2022-06-15T16:07:51.25453251Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"        \tTest:       \tTestGet\n"}
{"Time":"2022-06-15T16:07:51.254641135Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"--- FAIL: TestGet (0.00s)\n"}
{"Time":"2022-06-15T16:07:51.270588302Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"panic: runtime error: invalid memory address or nil pointer dereference [recovered]\n"}
{"Time":"2022-06-15T16:07:51.271907468Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"\tpanic: runtime error: invalid memory address or nil pointer dereference\n"}
{"Time":"2022-06-15T16:07:51.271983177Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x272e54]\n"}
{"Time":"2022-06-15T16:07:51.27199151Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"\n"}
{"Time":"2022-06-15T16:07:51.272055968Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"goroutine 19 [running]:\n"}
{"Time":"2022-06-15T16:07:51.272124052Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"testing.tRunner.func1.2({0x2b0780, 0x534be0})\n"}
{"Time":"2022-06-15T16:07:51.272160135Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"\t/usr/local/go/src/testing/testing.go:1389 +0x1c8\n"}
{"Time":"2022-06-15T16:07:51.27219301Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"testing.tRunner.func1()\n"}
{"Time":"2022-06-15T16:07:51.272228593Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"\t/usr/local/go/src/testing/testing.go:1392 +0x380\n"}
{"Time":"2022-06-15T16:07:51.272284343Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"panic({0x2b0780, 0x534be0})\n"}
{"Time":"2022-06-15T16:07:51.27232226Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"\t/usr/local/go/src/runtime/panic.go:838 +0x20c\n"}
{"Time":"2022-06-15T16:07:51.272365968Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"go-client/helper/rest/api/v2010.TestGet(0x0?)\n"}
{"Time":"2022-06-15T16:07:51.272400593Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"\t/go/src/twilio-go/helper/rest/api/v2010/api_test.go:33 +0xa4\n"}
{"Time":"2022-06-15T16:07:51.272565052Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"testing.tRunner(0x4000085a00, 0x32ba28)\n"}
{"Time":"2022-06-15T16:07:51.273228052Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"\t/usr/local/go/src/testing/testing.go:1439 +0x110\n"}
{"Time":"2022-06-15T16:07:51.273257927Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"created by testing.(*T).Run\n"}
{"Time":"2022-06-15T16:07:51.27334326Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":"\t/usr/local/go/src/testing/testing.go:1486 +0x328\n"}
{"Time":"2022-06-15T16:07:51.292365843Z","Action":"fail","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Elapsed":0}
{"Time":"2022-06-15T16:07:51.292463552Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Output":"FAIL\tgo-client/helper/rest/api/v2010\t0.074s\n"}
{"Time":"2022-06-15T16:07:51.292475052Z","Action":"fail","Package":"go-client/helper/rest/api/v2010","Elapsed":0.074}
{"Time":"2022-06-15T16:07:51.293718385Z","Action":"output","Package":"go-client/terraform/client","Output":"?   \tgo-client/terraform/client\t[no test files]\n"}
{"Time":"2022-06-15T16:07:51.293730843Z","Action":"skip","Package":"go-client/terraform/client","Elapsed":0}
{"Time":"2022-06-15T16:07:56.224970179Z","Action":"run","Package":"go-client/terraform/resources","Test":"TestCreateCall"}

Reply:
Looks like prism is returning an error:

client/helper/rest/api/v2010","Test":"TestGet","Output":" \tError Trace:\tapi_test.go:31\n"}
{"Time":"2022-06-15T16:07:51.254478468Z","Action":"output","Package":"go-client/helper/rest/api/v2010","Test":"TestGet","Output":" \tError: \tExpected nil, but got: \u0026url.Error{Op:"Get", URL:"http://prism_twilio:4010/v1/Credentials/AWS", Err:(*net.OpError)(0x4000184000)}\n"}

Reply:
Error I see locally:

{"Time":"2022-06-15T18:05:00.006401287Z","Action":"output","Package":"go-client/terraform/resources","Test":"TestCreateCredentialAws","Output":"    api_default.go:345: Unexpected call to *client.MockApiV2010.UpdateCredentialAws([CR123 0xc000188e00]) at /go/src/twilio-go/terraform/resources/api_default.go:345 because: there are no expected calls of the method \"UpdateCredentialAws\" for that receiver\n"}

Which makes sense since the Terraform code changed for some reason.

Reply:
Ah, the reason it changed is because the new `permissions` property is not something that can be set when the resource is created, so it must be _updated_ after creation, this the call to `updateCredentialsAWS`. If you add the property to the create operation in the test fixture this change should revert back.